packages/custom-elements/src/CustomElementRegistry.js

```diff
@@ -272,7 +271,7 @@ export default class CustomElementRegistry {
     // Resolve immediately only if the given local name has a definition *and*
     // the full document walk to upgrade elements with that local name has
     // already happened.
-    if (definition && !this._pendingDefinitions.some(d => d.localName === localName)) {
+    if (definition && this._unflushedLocalNames.indexOf(localName) === -1) {
```

Comment:
Simpler lookup if this were a Map.

Reply:
I changed this to a Set.

Reply:
Hmm. Let me try out the Map again though. I was trying to avoid having empty arrays sitting around before the flush but I'm not sure it's worth it after trying out Set.

Reply:
Nevermind, I think sticking with a Set here is reasonable.

Reply:
Welp, I flopped back...